File: src/backend/AgenticCompany.Core/Entities/Node.cs
using AgenticCompany.Core.Enums;

namespace AgenticCompany.Core.Entities;

public class Node
{
    public Guid Id { get; set; }
    public Guid? TenantId { get; set; }
    public Guid? ParentId { get; set; }
    public string Name { get; set; } = string.Empty;
    public NodeType Type { get; set; }
    public string? Description { get; set; }
    public string Path { get; set; } = string.Empty; // ltree-style path
    public int Depth { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public Node? Parent { get; set; }
    public ICollection<Node> Children { get; set; } = [];
    public ICollection<Principle> Principles { get; set; } = [];
    public ICollection<Spec> Specs { get; set; } = [];
    public ICollection<NodeMember> Members { get; set; } = [];
}

File: src/frontend/src/api/schemas/node.ts
import { z } from 'zod';

export const NodeType = z.enum(['Company', 'Organization', 'Squad', 'Team', 'Project']);
export type NodeType = z.infer<typeof NodeType>;

export interface Node {
  id: string;
  tenantId: string | null;
  parentId: string | null;
  name: string;
  type: NodeType;
  description: string | null;
  path: string;
  depth: number;
  createdAt: string;
  updatedAt: string;
  children?: Node[];
}

export const NodeSchema: z.ZodType<Node> = z.object({
  id: z.string().uuid(),
  tenantId: z.string().uuid().nullable(),
  parentId: z.string().uuid().nullable(),
  name: z.string().min(1),
  type: NodeType,
  description: z.string().nullable(),
  path: z.string(),
  depth: z.number().int(),
  createdAt: z.string().datetime(),
  updatedAt: z.string().datetime(),
  children: z.lazy(() => z.array(NodeSchema)).optional(),
});

export const CreateNodeSchema = z.object({
  parentId: z.string().uuid().nullable(),
  name: z.string().min(1, 'Name is required'),
  type: NodeType,
  description: z.string().nullable().optional(),
});

export type CreateNode = z.infer<typeof CreateNodeSchema>;
